### PR TITLE
Becoming a Traitor/Ninja is now an Offer, Ninja has a new crew antag type. 

### PR DESCRIPTION
**Pull request by nevimer**
## About The Pull Request
This pull request makes becoming a traitor (sleeper agent variety) an offer that you have 60 seconds to respond to. You can close the box if you refuse. 

This pull request also makes ninjas a crew antagonist under the same job-offer context - You can get an offer to be a ninja as crew (ghost ninja still exists). The goal is to make rolling these things as crew members far more likely to discourage observers sitting around for ghost role antagonists. 

## Why It's Good For The Game

As stated above, this should remove some load off admins for ahelps to remove antagonists from players, and should somewhat encourage players to be in round for certain antagonist roles. We want to encourage players to be in the game, not being ghosts waiting for midrounds. 

## Changelog

:cl:
add: Crew Traitor
qol: Traitor and Ninja are midround offers, instead of hard assignments. 
/:cl:

### CODE REVIEW

File: modular_zubbers/modules/pda_antagonist/ninja.dm [hunk 1]
```diff
@@ -0,0 +1,69 @@
+/datum/dynamic_ruleset/midround/from_living/ninja
+	name = "Crew Ninja"
+	offer_name = "Spider Clan Recruit"
+	midround_ruleset_style = MIDROUND_RULESET_STYLE_LIGHT
+	antag_datum = /datum/antagonist/ninja/crew
+	antag_flag = ROLE_NINJA
```
Comment (Majkl-J) [inline modular_zubbers/modules/pda_antagonist/ninja.dm:6]:
```suggestion
	antag_flag = ROLE_SLEEPER_NINJA
```

Make a new define for this one, keep the override as-is

**Review by Majkl-J — CHANGES_REQUESTED** [review]

**Comment by zeroisthebiggay** [comment]
how does crew ninja work
do you just
vanish if you say yes and get respawned

**Comment by StrangeWeirdKitten** [comment]
no

File: modular_zubbers/modules/pda_antagonist/ninja.dm (hunk 1 above)
Comment (StrangeWeirdKitten) [reply modular_zubbers/modules/pda_antagonist/ninja.dm:6]: ```FAILURE #1: /datum/dynamic_ruleset/midround/from_living/ninja has a non-unique antag_flag Space Ninja (used by /datum/dynamic_ruleset/midround/from_ghosts/space_ninja)! at code/modules/unit_tests/dynamic_ruleset_sanity.dm:40```

**Comment by StrangeWeirdKitten** [comment]
I'm going to break linters just so we can test merge this. It can be fixed before merge.

**Comment by projectkepler-ru** [comment]
@nevimer do this for contractor too?

**Comment by Detective-Google** [comment]
this is neat

**Comment by Detective-Google** [comment]
I like this concept but a couple worries do come to mind- namely, are there situations in which someone can roll crew ninja that they're for one reason or another effectively unable to do it-- missing an arm, being dragged around in cuffs by sec, et cetera. Are there systems in place to mitigate this? what happens if someone gets crew ninja and for whatever reason they get intercepted before they take the items out of the box- does this put ninja gear in the hands of a random validhunter?

**Comment by Detective-Google** [comment]
Also, shouldn't all the heads probably be protected from rolling this- as well as secmed?

**Comment by Swiftfeather** [comment]
Very stale

**Pull request closed without merging**